### DISCUSSION TITLE
pac connection create errors with saying no auth profiles, though there is and selected

### DISCUSSION THREAD

**Discussion by ameed** [microsoft/powerplatform-build-tools#559] [Q&A]
I tried a few times and also cleared and recreated auth profile and made sure to select it as well, however when creating connection using Service Principal, it still fails with saying "Error: No auth profiles were found on this computer. Please run 'pac auth create' to create one.". I have tried creating SP connection using all three options: (org url, instance id, unqiue name), yet get same error. 
<img width="1010" alt="image" src="https://github.com/microsoft/powerplatform-build-tools/assets/343796/e04732cf-c215-48d6-8d52-3267d6545c6f">

Am i missing something ?
following is version info:

PS C:\Users\ugigigi> pac help                                                                                                                                                    
Microsoft PowerPlatform CLI
Version: 1.27.6+g9f62afb
Online documentation: https://aka.ms/PowerPlatformCLI
Feedback, Suggestions, Issues: https://github.com/microsoft/powerplatform-build-tools/discussions

Usage: pac [admin] [application] [auth] [canvas] [catalog] [connection] [connector] [copilot] [data] [help] [modelbuilder] [org] [package] [pcf] [pipeline] [plugin] [powerpages] [solution] [telemetry] [test] [tool] [virtual-agent]

  admin                       Work with your Power Platform Admin Account
  application                 Commands for listing and installing available Dataverse applications from AppSource
  auth                        Manage how you authenticate to various services
  canvas                      Operating with Power Apps .msapp files
  catalog                     (Preview) Commands for working with Catalog in Power Platform
  connection                  (Preview) Commands for working with Dataverse connection.
  connector                   (Preview) Commands for working with Power Platform Connectors
  copilot                     Tools and utilities for copilot scenarios
  data                        Import and export data from Dataverse.
  help                        Show help for the Microsoft Power Platform CLI.
  modelbuilder                Code Generator for Dataverse APIs and Tables
  org                         Work with your Dataverse organization.
  package                     Commands for working with Dataverse package projects
  pcf                         Commands for working with Power Apps component framework projects
  pipeline                    Work with Pipelines
  plugin                      Commands for working with Dataverse plug-in class library
  powerpages                  Commands for working with Power Pages website.
  solution                    Commands for working with Dataverse solution projects
  telemetry                   Manage telemetry settings.
  test                        (Preview) Execution of automated tests for a Power App
  tool                        Power Platform tools that can be installed and launched.
  virtual-agent               Commands for working with Power Virtual Agent bots

**Comment by YannickRe** [comment]
As far as I know (or could test), it only works with auth profiles of type "Universal" (and not "Dataverse"), and it only works for auth profiles with a user account and not with an Application User.
It's a step in the right direction, but the different auth profiles and the SPN limitations that are different for each command currently make it pretty confusing 🙂. I'd like a bit more "in CLI guidance" around it, just telling me that this auth profile doesn't work for x command or so.

**Comment by deano808** [comment]
Hi @ameed 
I struggled with this yesterday and agreed with @YannickRe as it's just not possible (plus the documentation should be much better).
Using the Power Platform Tools extension for VS Code, there's a create auth profile action, that worked creating a profile of type UNIVERSAL.
I can now create SPN's using the CLI which was my objective.

**Reply by ameed** [reply]
Have been struggling with "pac connection create" ... I am still not able to create SPN connection in a non-default (not Universal) Dataverse environment

**Comment by RichardG91** [comment]
Hello @ameed @YannickRe  @deano808 
I'm trying to assign a user / group to an environment with a security role using "**pac admin assign-user**" and "**pac admin assign-group**" using a service principal on the connection.
But I'm getting the same error "_No auth profiles were found on this computer. Please run 'pac auth create' to create one_".

I've also created the Service Principal connection as Universal but then I get "_Original exception: AADSTS7000215: Invalid client secret provided_"

Do you know if there is a solution or another way to acomplish it?
Thanks
